tests/pytest_tests/system_tests/test_public_api.py

```diff
@@ -58,6 +58,25 @@ def test_run_from_tensorboard(runner, relay_server, user, api, copy_asset):
         assert len(uploaded_files) == 17
 
 
+def test_fetching_artifact_files(user, wandb_init):
+    project = "test"
+    run = wandb_init(entity=user, project=project)
+    artifact = wandb.Artifact("test-artifact", "test-type")
+    with open("boom.txt", "w") as f:
+        f.write("testing")
+    artifact.add_file("boom.txt", "test-name")
+    run.log_artifact(artifact, aliases=["sequence"])
+    artifact.wait()
+    run.finish()
+
+    # fetch artifact and its file successfully
+    artifact = Api().artifact(
+        name=f"{user}/{project}/test-artifact:v0", type="test-type"
+    )
+    boom = artifact.files()[0]
+    assert boom.name == "test-name"
```

Comment:
Since this is an API level test we might as well finish it out by adding
```suggestion
    assert boom.name == "test-name"
    artifact_path = artifact.download()
    file_path = os.path.join(artifact_path, boom.name)
    assert os.path.exists(file_path)
    assert open(file_path).read() == "testing"
```

Reply:
added!